### DISCUSSION TITLE
push files to DVC without splitting them into chunks

### DISCUSSION THREAD

**Discussion by somunslotus1** [treeverse/dvc#10545] [Help]
when I use dvc push data to minio,  I found that data strored on minio has seperated chunks, is there any config to set to push data to minio  not use chunk?

**Comment by skshetry** [comment]
DVC does not upload files in chunks.

**Reply by somunslotus1** [reply]
for example, I use dvc trace dataset.zip, I config the minio config and use "dvc push", when dvc pushed successfully, I check data on minio console, I found that I can not direct find dataset.zip file, the dataset.zip store in another way, can dvc push data to minio maintain originanl file? 
![image](https://github.com/user-attachments/assets/36b1a62c-05cb-4c36-8893-e43f9abb1e64)

**Reply by skshetry** [reply]
DVC is a content-addressable storage. The filenames are uploaded as the file's md5 hash, but it is split into two levels: first two characters are used for creating a directory, and the rest is kept as a filename.

For example, if you did `dvc add foo`, and `foo` had `22a1a2931c8370d3aeedd7183606fd7f` md5 hash,
it'll be uploaded to `<remote>/22/a1a2931c8370d3aeedd7183606fd7f`.

```
└── 22
    └── a1a2931c8370d3aeedd7183606fd7f
```

**Reply by skshetry** [reply]
Also see https://dvc.org/doc/user-guide/project-structure/internal-files#structure-of-the-cache-directory.
